crates/bevy_ecs/src/world/mod.rs

```diff
@@ -749,6 +758,14 @@ impl World {
             });
         }
     }
+
+    pub fn increment_global_system_counter(&self) -> u32 {
+        self.global_system_counter.fetch_add(1, Ordering::SeqCst) // TODO: can this be relaxed?
+    }
+
+    pub fn get_global_system_counter(&self) -> u32 {
+        self.global_system_counter.load(Ordering::Acquire) // FIXME: is this ordering correct?
+    }
```

Comment:
I'd love feedback on the atomic ordering

Reply:
@cart did you see this comment? I don't understand atomics well enough to advise.

Reply:
I believe these can both be "relaxed" as their ordering relative to atomic operations on other threads doesn't matter (and there are no same thread compiler-order ambiguities that I can think of). Systems that happen "in parallel" don't care who increments the counter first, and systems always use the value from `increment_global_system_counter` rather than reading `get_global_system_counter()`. And even if systems _did_ use get_global_system_counter(), I can't find a way that it would break our logic.

Reply:
@Davier says this has also been fixed. We can hunt for the commit if it matters.

Reply:
I changed `increment_global_system_counter` to `AcqRel` in 10082e8683098724ff7010e0ebad9bd7f5ecff20. I'm really not confident enough in my understanding of atomics to change either to `Relaxed` myself.